File: integration_tests/obc/boot_settings.py
from .obc_mixin import OBCMixin, command, decode_return


def _parse_boot_settings(s):
    lines = s.split('\n')
    m = dict(map(lambda l: l.split(':'), lines))

    return {
        'Primary': int(m['Primary boot slots'].strip(), 16),
        'Failsafe': int(m['Failsafe boot slots'].strip(), 16),
        'Counter': int(m['Boot counter'].strip()),
        'Confirmed': m['Boot confirmed'].strip() == "Yes"
    }


class BootSettingsMixin(OBCMixin):
    @decode_return(_parse_boot_settings)
    @command("boot_settings get")
    def boot_settings(self):
        pass

File: src/commands/boot_settings.cpp
#include <cstdint>
#include <cstring>
#include "boot/settings.hpp"
#include "obc_access.hpp"
#include "terminal/terminal.h"

void BootSettingsCommand(std::uint16_t argc, char* argv[])
{
    if (argc == 1 && strcmp(argv[0], "get") == 0)
    {
        GetTerminal().Printf("Primary boot slots: 0x%.2X\n", GetBootSettings().BootSlots());
        GetTerminal().Printf("Failsafe boot slots: 0x%.2X\n", GetBootSettings().FailsafeBootSlots());
        GetTerminal().Printf("Boot counter: %ld\n", GetBootSettings().BootCounter());
        GetTerminal().Printf("Last confirmed boot counter: %ld\n", GetBootSettings().LastConfirmedBootCounter());
    }
    else
    {
        GetTerminal().Puts("boot_settings <get>");
    }
}
